File: threedollar-domain/threedollar-domain-rds/src/testFixtures/java/com/depromeet/threedollar/domain/rds/user/domain/review/ReviewCreator.java
package com.depromeet.threedollar.domain.rds.user.domain.review;

import java.util.Optional;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import com.depromeet.threedollar.domain.rds.user.domain.TestFixture;

import lombok.AccessLevel;
import lombok.Builder;
import lombok.NoArgsConstructor;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
@TestFixture
public class ReviewCreator {

    @Builder
    public static Review create(
        @NotNull Long storeId,
        @NotNull Long userId,
        @NotNull String contents,
        @Nullable Integer rating,
        @Nullable ReviewStatus status
    ) {
        return Review.builder()
            .storeId(storeId)
            .userId(userId)
            .contents(contents)
            .rating(Optional.ofNullable(rating).orElse(3))
            .status(Optional.ofNullable(status).orElse(ReviewStatus.POSTED))
            .build();
    }

}

File: threedollar-domain/threedollar-domain-rds/src/testFixtures/java/com/depromeet/threedollar/domain/rds/user/domain/review/ReviewCreator.kt
package com.depromeet.threedollar.domain.rds.user.domain.review

import com.depromeet.threedollar.domain.rds.user.domain.TestFixture

@TestFixture
object ReviewCreator {

    @JvmOverloads
    @JvmStatic
    fun create(
        storeId: Long,
        userId: Long,
        contents: String,
        rating: Int,
        status: ReviewStatus = ReviewStatus.POSTED
    ): Review {
        return Review.builder()
            .storeId(storeId)
            .userId(userId)
            .contents(contents)
            .rating(rating)
            .status(status)
            .build()
    }

}
